pinot-core/src/main/java/org/apache/pinot/core/operator/filter/RangeIndexBasedFilterOperator.java

```diff
@@ -56,12 +56,12 @@ protected FilterBlock getNextBlock() {
 
     int firstRangeId;
     int lastRangeId;
-    if (_rangePredicateEvaluator instanceof OfflineDictionaryBasedRangePredicateEvaluator) {
+    if (_rangePredicateEvaluator instanceof SortedDictionaryBasedRangePredicateEvaluator) {
```

Comment:
is this change related to this API refactoring?

Reply:
Yes. We should determine the evaluator by whether the dictionary is sorted, instead of the instance of the dictionary.

Reply:
I mean I did not see any change to OfflineDictionaryBasedRangePredicateEvaluator nor SortedDictionaryBasedRangePredicateEvaluator in this PR. Is this an existing bug for range predicate?

Reply:
Oh, that is under the `RangeIndexBasedFilterOperator`

Reply:
Ah. Got it. You made changes in RangePredicateEvaluatorFactory.java.